packages/flame/lib/src/components/component.dart

```diff
@@ -642,9 +641,13 @@ class Component {
   /// that intersect with this ray, in the order from those that are closest to
   /// the user to those that are farthest.
   ///
-  /// The return value is an [Iterable] of `(component, point)` pairs, which
-  /// gives not only the components themselves, but also the points of
-  /// intersection, in their respective local coordinates.
+  /// The return value is an [Iterable] of components. If the [nestedPoints]
+  /// parameter is given, then it will also report the points of intersection
+  /// for each component in its local coordinate space. Specifically, the last
+  /// element in the list is the point in the coordinate space of the returned
+  /// component, the element before the last is in that component's parent's
+  /// coordinate space, and so on. The [nestedPoints] list must be growable and
+  /// modifiable.
```

Comment:
> list must be growable and modifiable

Can't we clone the list to avoid this type of side effect? Or does this gives a huge impact on perfomance?

If so, we could at least clone at on the first call?

Reply:
It is an output parameter, so the entire idea is that the list will be modified by the function so that the user would then be able to look at it.